src/auth/TokenValidationUtil.kt

```diff
@@ -0,0 +1,48 @@
+package no.nav.rekrutteringsbistand.statistikk.auth
+
+import io.ktor.config.ApplicationConfig
+import io.ktor.util.KtorExperimentalAPI
+import no.nav.rekrutteringsbistand.statistikk.utils.Cluster
+import no.nav.security.token.support.core.configuration.ProxyAwareResourceRetriever
+import no.nav.security.token.support.ktor.IssuerConfig
+import no.nav.security.token.support.ktor.TokenSupportConfig
+import java.net.URL
+
+@KtorExperimentalAPI
+data class TokenValidationConfig(
+    val config: ApplicationConfig,
+    val resourceRetriever: ProxyAwareResourceRetriever
+)
+
+class TokenValidationUtil {
+    @KtorExperimentalAPI
+    companion object {
+        fun tokenValidationConfig(
+            issuerConfig: IssuerConfig,
+            resourceRetriever: ProxyAwareResourceRetriever = ProxyAwareResourceRetriever(
+                System.getenv("HTTP_PROXY")?.let { URL(it) }
+            )
```

Comment:
Synes dette er litt rart så åpen for forbedringer.
Vanskelig å forklare så kan gjerne diskutere dette IRL.